src/java/htsjdk/samtools/metrics/MetricBase.java

```diff
@@ -84,7 +85,10 @@ public int hashCode() {
         int result = 0;
         for (final Field f : getClass().getFields()) {
             try {
-                result = 31 * result + f.get(this).hashCode();
+                Object value = f.get(this);
+                value = new FormatUtil().format(value); //format the value the way it will be written to disk
```

Comment:
I wouldn't do either of those things.  The various java.text.Format classes that FormatUtil relies on underneath do not provide any guarantees regarding thread-safety, and I know for a fact that at least the DateFormat classes are non-thread-safe, so wouldn't be surprised to find the same thing with Decimal/NumberFormat.